### PR TITLE
Fixes Issue #84

### PR DESCRIPTION
**Pull request by RatchetJ**
Fixes the unnecessary notifications of 50% completion for the first tier
of techs and also Code of Laws -> Issue #84. This also fixes the colors
of the messages, which were not always switching correctly.

### CODE REVIEW

File: UI/Screens/CivicsTree.lua [hunk 1]
```diff
@@ -1114,49 +1114,51 @@ end
...
-            elseif percentageNextTurn >= halfway and isCurrentBoosted == false and CQUI_halfwayNotified[currentCivicID] ~= true then
-                LuaEvents.CQUI_AddStatusMessage("The current Civic, " .. Locale.Lookup( civicName ) .. ", is one turn away from maximum Inspiration potential.", 10, STATUS_MESSAGE_CIVIC);
-                CQUI_halfwayNotified[currentCivicID] = true;
-            end
-        end
+      -- Is the current civic completed? -> Could be moved to the "OnCivicComplete" function
+      -- Else is it greater than 50% and has yet to be displayed?
+      if percentageToBeDone >= 1 then
+          LuaEvents.CQUI_AddStatusMessage("The Civic, " .. Locale.Lookup( civicName ) .. ", is completed.", 10, CQUI_STATUS_MESSAGE_CIVIC);
+      elseif percentageNextTurn >= halfway and isCurrentBoosted == false and CQUI_halfwayNotified[currentCivicID] ~= true and civicName ~= "LOC_CIVIC_CODE_OF_LAWS_NAME" then
```
Comment (chaorace) [inline UI/Screens/CivicsTree.lua]: I dislike this implementation for ignoring techs/civics. It's just not the most flexible solution if we ever want to be changing this behavior in the future. I feel that the better solution would be to instead add the ignored civic/tech IDs directly to the CQUI_halfwayNotified table upon instantiation

File: UI/Screens/TechTree.lua [hunk 2]
```diff
@@ -1036,33 +1036,35 @@ function OnLocalPlayerTurnBegin()
...
-            elseif percentageNextTurn >= halfway and isCurrentBoosted == false and CQUI_halfwayNotified[currentTechID] ~= true then
-                LuaEvents.CQUI_AddStatusMessage("The current Technology, " .. Locale.Lookup( techName ) .. ", is one turn away from maximum Eureka potential.", 10, STATUS_MESSAGE_TECHS);
-                CQUI_halfwayNotified[currentTechID] = true;
-            end
-        end
+          -- Is the current tech completed? -> Could be moved to the "OnResearchComplete" function
+          -- Else is it greater than 50% and has yet to be displayed?
+          if percentageToBeDone >= 1 then
+              LuaEvents.CQUI_AddStatusMessage("The Technology, " .. Locale.Lookup( techName ) .. ", is completed.", 10, CQUI_STATUS_MESSAGE_TECHS);
+          elseif percentageNextTurn >= halfway and isCurrentBoosted == false and CQUI_halfwayNotified[currentTechID] ~= true and techName ~= "LOC_TECH_POTTERY_NAME" and techName ~= "LOC_TECH_ANIMAL_HUSBANDRY_NAME" and techName ~= "LOC_TECH_MINING_NAME" then
```
Comment (chaorace) [inline UI/Screens/TechTree.lua]: I dislike this implementation for ignoring techs/civics. It's just not the most flexible solution if we ever want to be changing this behavior in the future. I feel that the better solution would be to instead add the ignored civic/tech IDs directly to the CQUI_halfwayNotified table upon instantiation

File: UI/Screens/CivicsTree.lua (hunk 1 above)
Comment (RatchetJ) [inline UI/Screens/CivicsTree.lua]: Sure thing, that sounds like a good idea. Question: After I've made the changes, do I just do another pull request, or is there a way for me to edit this one?

**Comment by chaorace** [comment]
Works flawlessly, merging now! Thanks as always for the good work 👍
